accepted/0000-workspace-dependency-sharing.md

```diff
@@ -0,0 +1,343 @@
+# Limit dependencies shared between workspaces
+
+## Summary
+
+For all workspaces within a project, the default behavior of dependency
+resolution will be as follows:
+
+- Dependencies on other workspaces within the project will _always_ be
+  resolved to the corresponding sibling workspace.
+- Peer dependencies of multiple workspaces within the project will always
+  be shared with one another, raising an `ERESOLVE` if this is impossible.
```

Comment:
so this is explicitly saying that if you have 100 child workspaces, and 1 declares react as a peer dep, the other 99 have access to that version of react, and can't peer-depend on a conflicting version of react? What if they non-peer-depend on one? (the latter case might be a non-issue, if the peer's availability is due to hoisting and the child's react dependency is not hoisted per the next bullet point)

Reply:
Yeah, all peer deps across all workspaces would have to be resolvable (as they must be today).  But non-peer deps would be nested within the workspace, so no relevant conflicts.

Reply:
A workaround here (which also does work today) is to list a different version of the peerDep in the workspace's `devDependencies` for use in development.  This allows it to nest today, though it won't be default.  With this proposal, its presence in `devDependencies` would _force_ it to nest.